src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -580,11 +574,15 @@ protected List<File> scanFileSets() {
      * @return A list of files, retrieved from the given scanner.
      */
     private List<File> retrieveAllScannedFiles(DirectoryScanner scanner, int logIndex) {
-        final String[] fileNames = scanner.getIncludedFiles();
```

Comment:
Issue #16421: `S1488`: Local variables should not be declared and then immediately returned or thrown